### PR TITLE
Add missing attributes to listing

### PR DESCRIPTION
**Pull request by joaodubas**
Add the following attributes to a listing:

* `orientation`
* `sun period`
* `floor count`
* `unit per floor`
* `construction year`
* `price per area`

For each field a new filter was added:

* `orientation`
* `sun_period`
* `mix_floor_count` and `max_floor_count`
* `min_unit_per_floor` and `max_unit_per_floor`
* `min_age` and `max_age`
* `min_price_per_area` and `max_price_per_area`

GraphQL queries and mutations were adjusted in PR #484.

### CODE REVIEW

File: apps/re/lib/listings/schemas/listing.ex [hunk 1]
```diff
@@ -71,6 +78,10 @@ defmodule Re.Listing do
 
   @garage_types ~w(contract condominium)
 
+  @orientation_types ~w(frente lateral fundo meio)
```
[the remark this answers is not in the document]
Comment (joaodubas) [reply apps/re/lib/listings/schemas/listing.ex]: Suggestions here are welcome, I really tried to find what terms to use.

File: apps/re/lib/filtering/filtering.ex [hunk 2]
```diff
@@ -237,5 +249,80 @@ defmodule Re.Filtering do
...
+  end
+
+  defp attr_filter({:max_unit_per_floor, unit_per_floor}, query) do
+    from(
+      l in query,
+      where: l.unit_per_floor <= ^unit_per_floor
+    )
+  end
+
+  defp attr_filter({:orientation, orientation}, query) do
```
[the remark this answers is not in the document]
Comment (joaodubas) [reply apps/re/lib/filtering/filtering.ex]: :ok: will adjust both.

File: apps/re/test/listings/listings_test.exs [hunk 3]
```diff
@@ -406,6 +406,34 @@ defmodule Re.ListingsTest do
       assert {:ok, listing} = Listings.insert(@insert_listing_params, address, user)
       assert Repo.get(Listing, listing.id)
     end
+
+    test "should insert extra info for admin user" do
```
[the remark this answers is not in the document]
Comment (joaodubas) [reply apps/re/test/listings/listings_test.exs]: I was just checking what was added in this PR, will improve the description, or maybe, add these into an existing test.

**Comment by joaodubas** [comment]
> So, a couple of things:
> `sun period` and `orientation` seems like could be a tag, like `morning_sun` or `front_orientation`.
> I think for all attributes that are enum-like we can consider like tags.

The problem with tags, is that we can't validate that only one value from the list is possible, we could add this type of validation at the expense of making the tagging system more complex.

> `price per area` is calculated, why do we need a persisted field? If we need it persisted, we should calculate on change and not be manually filled.

The `price_per_area` isn't filled manually, it's calculated here:
https://github.com/emcasa/backend/blob/366c2f884a4884d6c66e4f821a08596db7d443c1/apps/re/lib/listings/schemas/listing.ex#L181-L191

Despite being a calculated field, it's easier to filter and order by when it's set as a column. Maybe in postgres 12 we'll have a feature to generate this type of column on the database side with [`generated columns`](https://wiki.postgresql.org/wiki/Todo#Data_Types).

File: apps/re/lib/listings/schemas/listing.ex (hunk 1 above)
[the remark this answers is not in the document]
Comment (joaodubas) [reply apps/re/lib/listings/schemas/listing.ex]: I think we can use: `frontside`, `backside`, `lateral`, `middle` or `inside`

[the remark this answers is not in the document]
Comment (pmargreff) [reply apps/re/lib/listings/schemas/listing.ex]: I guess these are really good, to the last one you can use `internal` as well.

**Pull request merged**